CMakeLists.txt

```diff
@@ -275,6 +275,10 @@ if(NOT MAKE_DOCS_ONLY)
     endif()
 
     if(MAKE_DOCS)
```

Comment:
why not remove L:277-281 and move 
 "message(STATUS "Only making documents")"
 "add_subdirectory(docs)"
outside of the conditional starting on L:66 ?

so... build the binary  if(NOT MAKE_DOCS_ONLY) 
and then check if MAKE_DOCS is set and output the status message and add_subdirectory(docs).